runtime/databricks/automl_runtime/forecast/model.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (C) 2022 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from abc import ABC, abstractmethod
+from typing import List
+
+import pandas as pd
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.models.signature import infer_signature, ModelSignature
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+class ForecastModel(ABC, mlflow.pyfunc.PythonModel):
+    @abstractmethod
+    def __init__(self):
+        super().__init__()
+
+    @property
+    @abstractmethod
+    def model_env(self):
+        pass
+
+    @staticmethod
+    def _validate_cols(df: pd.DataFrame, required_cols: List[str]):
+        df_cols = set(df.columns)
+        required_cols_set = set(required_cols)
+        if not required_cols_set.issubset(df_cols):
+            raise MlflowException(
+                message=(
+                    f"Input data columns '{list(df_cols)}' do not contain the required columns '{required_cols}'"
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+    def infer_signature(self, sample_input: pd.DataFrame = None) -> ModelSignature:
+        signature = infer_signature(sample_input, self.predict(context=None, model_input=sample_input))
```

Comment:
Why do we use `predict` instead of `predict_timeseries` here? I think calling `predict` with `model_input=None` will fail, at least for the `ArimaModel`. Could we add unit tests for `infer_signature` to verify if it work as expected?
Also I did see the example notebook where the model works for inference with pyspark dataframe. Does it mean that it actually works without the signature?

Reply:
To goal to add the infered signature is to make `predict` function work from mlflow side. We need to use `predict` to get the right output format.

Reply:
Oh I see. I missed the `sample_input` in the line. It makes sense then that we call `predict` to get a sample prediction. 
Is there a way to test in unit tests if the `predict` function work from mlflow side?

Reply:
mlflow has that unit test, I don't think we need to verify it on our side. We only need to make sure our predict function works properly and assume that `infer_signature` works.